### PR TITLE
[NUI] TextLayout can consider specified maximum width. 

### PR DESCRIPTION
**Pull request by rabbitfor**
### Description of Change ###
Previsouly, the Layout system did not reflect TextLabel's maximum size, so the following example did't work corrrectly,

```C#
  var view = new View()
  {
    Layout = new LinearLayout() { LinearOrientation = LinearLayout.Orientation.Horizontal, },
  };
  view.Add(new TextLabel() { MaximumSize = new Size2D(50, 50), Text = "Helloooooooooooooooooo!" });
  view.Add(new TextLabel() { MaximumSize = new Size2D(50, 50), Text = "Helloooooooooooooooooo!" });
  NUIApplication.GetDefaultWindow().Add(view);
```

The x-position of the second child was calculated based on the first child's natutal width,
but the first child's actual width was limited by MaximumSize(50).

So this patch update the TextLabel's layout to make it consider the maxmimum size of the view.

Signed-off-by: Jiyun Yang <ji.yang@samsung.com>

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs [hunk 1]
```diff
@@ -49,17 +49,20 @@ protected override void OnMeasure(MeasureSpecification widthMeasureSpec, Measure
...
                 {
+                    var minSize = Owner.MinimumSize;
+                    var maxSize = Owner.MaximumSize;
+                    var naturalSize = Owner.GetNaturalSize();
+
                     if (heightMeasureSpec.Mode == MeasureSpecification.ModeType.Exactly)
                     {
                         // GetWidthForHeight is not implemented.
-                        totalWidth = Owner.GetNaturalSize().Width;
+                        totalWidth = Math.Min(Math.Max(naturalSize.Width, minSize.Width), (maxSize.Width < 0 ? Int32.MaxValue : maxSize.Width));
```
Comment (huiyueun) [inline src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs:59]: minSize나 maxSize는 보통 사용자가 설정할 경우에만 있는 값인가요?

Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/TextLabel.cs:59]: 사용자가 설정하지 않을경우, minSize 는 0, maxSize 는 엄청난 음수로 조회됩니다.

**Review by huiyueun — COMMENTED** [review]
LGTM

**Review by huiyueun — APPROVED** [review]

**Comment by jaehyun0cho** [comment]
Do we need to consider widthMeasureSpec.Mode == MeasureSpecification.ModeType.Exactly?

i.e. should we also consider max height with totalHeight = Owner.GetHeightForWidth(totalWidth); in line 46 ?

**Comment by rabbitfor** [comment]
> Do we need to consider the case that widthMeasureSpec.Mode == MeasureSpecification.ModeType.Exactly?
> 
> i.e. should we also consider max height with totalHeight = Owner.GetHeightForWidth(totalWidth); in line 46 ?

We need to test for the cases you mentioned. (This patch does only care `Unspecified` mode.)

**Pull request merged**